pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -107,3 +108,31 @@ def attach_model_logging_functions(self, model):
         for callback in self.trainer.callbacks:
             callback.log = model.log
             callback.log_dict = model.log_dict
+
+    def _attach_model_callbacks(self, model: LightningModule) -> None:
+        """
+        Attaches the callbacks defined in the model.
+        If a callback returned by the model's configure_callback method has the same type as one or several
+        callbacks already present in the trainer callbacks list, it will replace them.
+
+        Args:
+            A model which may or may not define new callbacks in
+            :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_callbacks`.
+        """
+        model_callbacks = model.configure_callbacks()
```

Comment:
Can this merging function be a static method ?

Reply:
hmm, not so easy. need access to `self` to call `self._reorder_callbacks`

Reply:
You could do `ModelConnector._reorder_callbacks`, having everything static :)

Reply:
done that